flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/KafkaTestBase.java

```diff
@@ -270,6 +270,42 @@ public static <K, V> void produceToKafka(
         }
     }
 
+    public static <K, V> void produceTransactionalMessageToKafka(
```

Comment:
Instead of code replication, how about patching `produceToKafka` to create transactions if transaction.id is set?

Reply:
I agree, I've applied your idea. Thanks for the input